R/pandoc.R

```diff
@@ -551,34 +551,58 @@ unix_mathjax_path <- function() {
 
 
 pandoc_html_highlight_args <- function(template,
-                                       highlight) {
+                                       highlight,
+                                       highlight_downlit = FALSE) {
 
   args <- c()
 
-  if (is.null(highlight)) {
-    args <- c(args, "--no-highlight")
-  }
-  else if (!identical(template, "default")) {
-    if (identical(highlight, "default"))
-      highlight <- "pygments"
-    args <- c(args, "--highlight-style", highlight)
+  # no highlighting engine
+  if (is.null(highlight)) return(pandoc_highlight_args(NULL))
+
+  # TODO: move out so that it works also for other formats
+  resolve_highlight <- function(highlight) {
+    custom <- list(
+      # from distill
+      # https://github.com/rstudio/distill/blob/c98d332192ff75f268ddf69bddace34e4db6d89b/inst/rmarkdown/templates/distill_article/resources/a11y.theme
+      a11y = pkg_file_highlight("a11y.theme"),
+      # from distill
+      # https://github.com/rstudio/distill/blob/c98d332192ff75f268ddf69bddace34e4db6d89b/inst/rmarkdown/templates/distill_article/resources/rstudio.theme
+      rstudio = pkg_file_highlight("rstudio.theme")
+    )
+    custom[[highlight]] %||% highlight
   }
-  else {
-    highlight <- match.arg(highlight, html_highlighters())
+  highlight <- resolve_highlight(highlight)
+
+  # downlit engine
+  if (highlight_downlit) {
     if (is_highlightjs(highlight)) {
-      args <- c(args, "--no-highlight")
-      args <- c(args, "--variable", "highlightjs=1")
-    }
-    else {
-      args <- c(args, "--highlight-style", highlight)
+      stop(
+        sprintf(
+          "%s theme is for highlightjs highlighting engine",
+          "and can't be used with downlit.",  highlight
+        ),
+        call. = FALSE
+      )
     }
+    args <- c(
+      pandoc_highlight_args(highlight, default = resolve_highlight("a11y")),
+      pandoc_variable_arg("highlight-downlit")
```

Comment:
It might worth a comment that these variables are included primarily for testing?

Reply:
For testing ? It is not only for testing.

`highlight-downlit` is a pandoc variable I need to introduced in order to have a piece of css inserted (same mechanism as for highlightjs). 

Here : https://github.com/rstudio/rmarkdown/pull/1941/files#diff-91b59569aab1b937e1e5a6e1559a62de19e04f4c4bad52bfb08b910281e0f2dfR65

The css is for having the color scheme of highlighting style applies on links created by downlit. Otherwise, they are like any other hyperlinks (blue and underlined)

It is a small piece of css but it seemed necessary to a good visual output.

but I could add a comment about this.

Reply:
Ah ok, I missed that. I think it's probably important to note somewhere that `downlit_highlight` will insert  new css, but it's not clear to me where that should be.

Reply:
It may be worth adding a section on highlighting in the help page. That would allow to give this sort of details about 

* highlightjs which insert css and js in the template
* Custom theme included in the package in `.theme` file
* downlit which will replace pandoc highlighting modifying the html nodes and inserting css in the template

It is better than nowhere.

Reply:
Added a whole section on Highlighting if you want to look at it: https://github.com/rstudio/rmarkdown/pull/1941/files#diff-93d1e6747a8a5a21e4668d9a67b3889d481426747e8a0f18d7b2ebfb32ee8413R101